specs/interop/honest-nodes.md

```diff
@@ -0,0 +1,183 @@
+# Honest Nodes
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Sequencing](#sequencing)
+- [Unsafe](#unsafe)
+- [Cross-Unsafe](#cross-unsafe)
+- [Local Safe](#local-safe)
+  - [Magic `derive` Function](#magic-derive-function)
+  - [Task Processing Details](#task-processing-details)
+    - [No Derivation Required](#no-derivation-required)
+    - [L1 Canonical Checks](#l1-canonical-checks)
+    - [Normal Blocks](#normal-blocks)
+    - [Deposit Only Blocks](#deposit-only-blocks)
+    - [Deposit-only Bisection Process](#deposit-only-bisection-process)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+Defines one possible implementation of a sequencing or verifying node that uses the [supervisor API] to perform
+efficient verification of executing message dependencies. Implementations MAY differ from this approach providing they
+still adhere to the protocol rules.
+
+## Sequencing
+
+Sequencing nodes continue building new `unsafe` blocks on top of the existing `unsafe` chain head. For
+each [ExecutingMessage event][event] emitted by a transaction being added to the new block, the sequencer MUST
+call [`interop_checkMessage`] with the identifier and payload hash from the event.
+
+Sequencers MUST remove any transactions with any `conflicts` or `unknown` events from the block.
```

Comment:
It could be good to use language that implies "drop transactions while building the block" to make it clear its not removing any transactions from "finalized" blocks